src/umap.cpp

```diff
@@ -170,9 +171,9 @@ void umap_cfg_set_bufsize( int page_bufsize )
 }
 
 //--------------------------for multi-file support----------------------
-_umap::_umap(void* _mmap_addr, size_t _mmap_length, int num_backing_file,umap_backing_file* backing_files)
+_umap::_umap(void* _mmap_addr, size_t _mmap_length,vector<umap_backing_file>* backing_files)
     :   segment_address{_mmap_addr}, segment_length{_mmap_length},
-	      num_bk_files{num_backing_file},bk_files{backing_files},
+	bk_files{*backing_files},
```

Comment:
remove this initializer and add the following to the constructor implementation:
```C++
for (umap_backing_file* bf = &backing_files[0]; bf < &backing_files[num_backing_file]; ++bf)
    bk_files.pushback(umap_backing_file{bf->fd, bf->data_size, bf->data_offset});
```